components/apply-section/deadline/deadline.js

```
@@ -46,22 +48,18 @@ export default function Deadline() {
     }
   )
 
-  const newClassStartDate = new Date(classStart).toLocaleString('en', {
-    day: 'numeric',
-    month: 'long',
-    year: 'numeric'
-  })
-
-
-  return (
-    <React.Fragment>
-      <Typography className={classes.deadline}>
-        <span className={classes.newClassNumber}>Class {newClassNumber}</span> starts on {newClassStartDate} | Application Deadline: {applicationEndDate}
-      </Typography>
-      <Container>
-        <Timer date={applicationEndDate} />
-      </Container>
-    </React.Fragment>
-  )
-
+  if (applicationDeadline) {
```

I would recreate the flow of this function so it starts like this:
```
if(!applicationDeadline) {
  return;
}
```
This is called fail fast. So take care of the case where the function should not do anything to begin with (then that is out of mind) And you can then focus on what matters (the functionality that actually matters).

What do you think of that idea?